osquery/tables/events/darwin/openbsm_events.cpp

```diff
@@ -156,10 +197,109 @@ Status OpenBSMExecVESubscriber::Callback(
     }
   }
   r["uptime"] = INTEGER(tables::getUptime());
+
+  try {
+    r["parent"] = INTEGER(ppid_map.at(pid));
+  } catch (std::out_of_range) {
+    /* Mapping doesn't exist so no fork was captured. Not fatal. Ignoring. */
+  }
+
   add(r);
   return Status(0, "OK");
 }
 
+Status OpenBSMProcEventsSubscriber::handleFork(
+    const OpenBSMEventContextRef& ec) {
+  uint32_t ppid = 0; // Parent PID
+  uint32_t cpid = 0; // Child PID
+
+  for (const auto& tok : ec->tokens) {
+    switch (tok.id) {
+      /* The parent is the process issuing the syscall. Its PID is given on the
+       * subject token.
+       */
+      case AUT_SUBJECT32:
+        ppid = tok.tt.subj32.pid;
+        break;
+      case AUT_SUBJECT64:
+        ppid = tok.tt.subj64.pid;
+        break;
+      case AUT_SUBJECT32_EX:
+        ppid = tok.tt.subj32_ex.pid;
+        break;
+      case AUT_SUBJECT64_EX:
+        ppid = tok.tt.subj64_ex.pid;
+        break;
+
+      /* Child PID is given as the first argument here. This could also be
+       * extract from the return token as the child PID should be the return
+       * value of the fork syscall given to the parent.
+       */
+      case AUT_ARG32:
+        if (tok.tt.arg32.no == 0) {
```

Comment:
Nit. Ternary here.

Reply:
Not sure I get it? Do you mean `cpid = tok.tt.arg32.no == 0 ? tok.tt.arg32.val : 0`?

Reply:
Yeah, generally single set ifs are nice as ternarys

Reply:
Hum, but here I'm not assigning for the case `tok.tt.arg32.no != 0` that's why I didn't do it. However if this is standard in the codebase I have no problem with it.